controller/api/models.py

```diff
@@ -353,6 +367,58 @@ def _start_containers(self, to_add):
         if set([c.state for c in to_add]) != set(['up']):
             err = 'warning, some containers failed to start'
             log_event(self, err, logging.WARNING)
+        # if the user specified a health check, try checking to see if it's running
+        try:
+            config = self.config_set.latest()
+            if 'HEALTHCHECK_URL' in config.values.keys():
+                self._healthcheck(to_add, config.values)
+        except Config.DoesNotExist:
+            pass
+
+    def _healthcheck(self, containers, config):
+        # if at first it fails, back off and try again at 10%, 50% and 100% of INITIAL_DELAY
+        intervals = [1.0, 0.1, 0.5, 1.0]
+        # HACK (bacongobbler): we need to wait until publisher has a chance to publish each
+        # service to etcd, which can take up to 20 seconds.
+        time.sleep(20)
+        for i in range(0, 4):
+            delay = int(config.get('HEALTHCHECK_INITIAL_DELAY', 0))
+            try:
+                # sleep until the initial timeout is over
+                if delay > 0:
+                    time.sleep(delay * intervals[i])
+                self._do_healthcheck(containers, config)
+                break
+            except exceptions.HealthcheckException as e:
+                try:
+                    new_delay = delay * intervals[i+1]
```

Comment:
@Blystad to explain with a unit test run:

```
$ cat logs/newish-zoologer.log
2015-08-03T19:26:07UTC deis[api]: autotest created initial release
2015-08-03T19:26:07UTC deis[api]: autotest deployed autotest/example
2015-08-03T19:26:07UTC deis[api]: autotest scaled containers cmd=1
2015-08-03T19:26:07UTC deis[api]: autotest added HEALTHCHECK_URL
2015-08-03T19:26:07UTC deis[api]: app failed health check (got '404', expected: '200'); trying again in 0.0 seconds
2015-08-03T19:26:07UTC deis[api]: app failed health check (got '404', expected: '200'); trying again in 0.0 seconds
2015-08-03T19:26:07UTC deis[api]: app failed health check (got '404', expected: '200'); trying again in 0.0 seconds
2015-08-03T19:26:07UTC deis[api]: app failed health check (got '404', expected: '200')
2015-08-03T19:26:07UTC deis[api]: aborting, app containers failed to respond to health check
```

the try/catch block just wraps the call to `log_event`. The fourth healthcheck will still occur as expected. It's just for that fancy `trying again in x seconds` message.